From: Akihiro Sagawa <sagawa.aki@gmail.com>
Subject: [PATCH resend] win32u: Make elfFullName from family name in the font replacement.
Message-Id: <20211118224937.1F2A.375B48EC@gmail.com>
Date: Thu, 18 Nov 2021 22:49:53 +0900


The original Kusanagi code, commit eab02574929d, was implemented in this way.
However, commit abc8f2c6c270 and f360057e2f99 made it from face->full_name.

Signed-off-by: Akihiro Sagawa <sagawa.aki@gmail.com>
---
 dlls/win32u/font.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/dlls/win32u/font.c b/dlls/win32u/font.c
index 11111111111..11111111111 100644
--- a/dlls/win32u/font.c
+++ b/dlls/win32u/font.c
@@ -3128,7 +3128,14 @@ static BOOL enum_face_charsets( const struct gdi_font_family *family, struct gdi
     if (family != face->family)
     {
         lstrcpynW( elf.elfLogFont.lfFaceName, family->family_name, LF_FACESIZE );
-        lstrcpynW( elf.elfFullName, face->full_name, LF_FULLFACESIZE );
+        lstrcpynW( elf.elfFullName, family->family_name, LF_FULLFACESIZE );
+        if (!(face->ntmFlags & NTM_REGULAR) &&
+            lstrlenW( elf.elfFullName ) + lstrlenW( face->style_name ) + 1 < LF_FULLFACESIZE)
+        {
+            static const WCHAR spaceW[] = {' ',0};
+            lstrcatW( elf.elfFullName, spaceW );
+            lstrcatW( elf.elfFullName, face->style_name );
+        }
     }
     if (subst) lstrcpynW( elf.elfLogFont.lfFaceName, subst, LF_FACESIZE );
 

